File: ios/dartchat/RNUserDefaults.m
//
//  RNUserDefaults.m
//  dartchat
//
//  Created by Siddharth Hathi on 9/4/23.
//

#import <Foundation/Foundation.h>
#import "React/RCTBridgeModule.h"

// THIS MODULE DOES NOT WORK -> SWIFT BRIDGE FAILS
// SEE objc_rn.m FOR WORKING OBJC BRIDGE

@interface RCT_EXTERN_MODULE(RNUserDefaults, NSObject)

//RCT_EXTERN_METHOD(validate: (NSString *)validationString)

RCT_EXTERN_METHOD(
  storeData: (NSString*)key
  val:(NSString*)val
)

@end

File: ios/dartchat/RNUserDefaults.swift
//
//  RNUserDefaults.swift
//  dartchat
//
//  Created by Siddharth Hathi on 9/4/23.
//

import React

@objc(RNUserDefaults)
public class RNUserDefaults: NSObject {
  @objc(: val:) func storeData(_ key: String, val: String) -> Void {
    UserDefaults.standard.set(val as String, forKey: key as String);
    NSLog("data set in user default store");
  }
  
  @objc
  func validate(_ validationString: String?) -> Void {
    NSLog(validationString ?? "No validation string");
    NSLog("native function called")
  }
  
  @objc
  static func requiresMainQueueSetup() -> Bool {
    return true
  }
}
